src/lib.rs

```diff
@@ -6,70 +6,265 @@
 //! Turn Traditional Chinese script to Simplified Chinese script and vice-versa. Check string script to determine if string is Traditional or Simplified Chinese.
 //!
 //! ### Usage
-//! ```rust
+//! ```
 //! extern crate character_converter;
 //!
-//! use character_converter::CharacterConverter;
+//! use character_converter::*;
 //!
-//! let converter: CharacterConverter = CharacterConverter::new();
 //!
-//! let traditional_text = "欧洲";
-//! let simplified_text = "歐洲";
+//! let traditional_text = "歐洲";
+//! let simplified_text = "欧洲";
 //!
 //! // Check script
-//! let result_one: bool = converter.is_traditional(traditional_text);
-//! println!("{}", result_one); // --> true
+//! assert!(is_traditional(traditional_text));
 //!
-//! let result_two: bool = converter.is_simplified(traditional_text);
-//! println!("{}", result_two); // --> false
+//! assert!(!is_simplified(traditional_text));
 //!
 //! // Convert script
-//! let result_three: String = converter.traditional_to_simplified(traditional_text);
-//! println!("{}", result_three == simplified_text); // --> true
+//! let result_three = traditional_to_simplified(traditional_text);
+//! assert_eq!(result_three, simplified_text);
 //!
-//! let result_four: String = converter.simplified_to_traditional(simplified_text);
-//! println!("{}", result_four == traditional_text); // --> true
+//! let result_four = simplified_to_traditional(simplified_text);
+//! assert_eq!(result_four, traditional_text);
+//! ```
+#![cfg_attr(feature = "bench", feature(test))]
 
 extern crate bincode;
 
-mod character_converter;
-pub use self::character_converter::Converter as CharacterConverter;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use bincode::deserialize_from;
+use fst::raw::{Fst, Output};
+use once_cell::sync::Lazy;
+
+static T2S: Lazy<HashMap<String, String>> =
+    Lazy::new(|| deserialize_from(&include_bytes!("../data/t2s.profile")[..]).unwrap());
+static S2T: Lazy<HashMap<String, String>> =
+    Lazy::new(|| deserialize_from(&include_bytes!("../data/s2t.profile")[..]).unwrap());
+
+// create an fst containing all the keys
+static T2S_FST: Lazy<Fst<Vec<u8>>> = Lazy::new(|| {
+    let mut keys: Vec<_> = T2S.keys().collect();
+    keys.sort_unstable();
+    Fst::from_iter_set(keys).unwrap()
+});
+
+// create an fst containing all the keys
+static S2T_FST: Lazy<Fst<Vec<u8>>> = Lazy::new(|| {
+    let mut keys: Vec<_> = S2T.keys().collect();
+    keys.sort_unstable();
+    Fst::from_iter_set(keys).unwrap()
+});
+
+fn is_script(
+    raw: &str,
+    mapping: &HashMap<String, String>,
+    backup: &HashMap<String, String>,
+) -> bool {
+    let mut buffer = [0; 4];
+    for character in raw.chars() {
+        let c = character.encode_utf8(&mut buffer);
+        if mapping.get(c).is_none() {
+            if backup.get(c).is_some() {
```

Comment:
Funny thing, below suggestion is slower:
```suggestion
        if mapping.contains_key(c) {
            if backup.contains_key(c) {
```

time diff:
```diff
// original code
+ test benches::bench_simplified_is_simplified   ... bench:         845 ns/iter (+/- 32)
+ test benches::bench_simplified_is_traditional  ... bench:         188 ns/iter (+/- 1)
  test benches::bench_simplified_to_simplified   ... bench:       2,260 ns/iter (+/- 18)
  test benches::bench_simplified_to_traditional  ... bench:       2,571 ns/iter (+/- 34)
+ test benches::bench_traditional_is_simplified  ... bench:         170 ns/iter (+/- 2)
+ test benches::bench_traditional_is_traditional ... bench:         787 ns/iter (+/- 6)
  test benches::bench_traditional_to_simplified  ... bench:       2,561 ns/iter (+/- 44)
  test benches::bench_traditional_to_traditional ... bench:       2,211 ns/iter (+/- 21)

// suggestion
- test benches::bench_simplified_is_simplified   ... bench:         978 ns/iter (+/- 23)
- test benches::bench_simplified_is_traditional  ... bench:         217 ns/iter (+/- 3)
  test benches::bench_simplified_to_simplified   ... bench:       2,267 ns/iter (+/- 16)
  test benches::bench_simplified_to_traditional  ... bench:       2,570 ns/iter (+/- 33)
- test benches::bench_traditional_is_simplified  ... bench:         223 ns/iter (+/- 4)
- test benches::bench_traditional_is_traditional ... bench:         949 ns/iter (+/- 30)
  test benches::bench_traditional_to_simplified  ... bench:       2,548 ns/iter (+/- 39)
  test benches::bench_traditional_to_traditional ... bench:       2,189 ns/iter (+/- 76)
```

Reply:
Wow, that is quite strange indeed. But I would say that we should keep the `contains_key` version as it better expresses what we want to do instead of searching for performance, here.

Reply:
Agree